plugins/inputs/netdata/netdata.go

```diff
@@ -0,0 +1,291 @@
+	Family     string                    `json:"family"`
+	DataURL    string                    `json:"data_url"`
+	Dimensions map[string]ChartDimension `json:"dimensions"`
+}
+
+type ChartDimension struct {
+	Name string `json:"name"`
+}
+
+type ResponseData struct {
+	DimensionNames []string   `json:"dimension_names"`
+	Result         DataResult `json:"result"`
+}
+
+type DataResult struct {
+	Data [][]interface{} `json:"data"`
+}
+
+func init() {
+	inputs.Add("netdata", func() telegraf.Input {
+		return &Netdata{
+			Points: 10,
+			Group:  "average",
+			PrefixChartIDByHostname: true,
+		}
+	})
+}
+
+func (n *Netdata) Description() string {
+	return description
+}
+
+func (n *Netdata) SampleConfig() string {
+	return sampleConfig
+}
+
+func (n *Netdata) Gather(acc telegraf.Accumulator) error {
+	var wg sync.WaitGroup
+
+	if n.client == nil {
+		tr := &http.Transport{
+			ResponseHeaderTimeout: time.Duration(3 * time.Second),
+		}
+		n.client = &http.Client{
+			Transport: tr,
+			Timeout:   time.Duration(4 * time.Second),
+		}
+	}
+
+	for idx := range n.Servers {
+		wg.Add(1)
+		go func(server *Server) {
+			defer wg.Done()
+			n.gatherServer(acc, server)
+		}(&n.Servers[idx])
+	}
+
+	wg.Wait()
+	return nil
+}
+
+func (n *Netdata) gatherServer(acc telegraf.Accumulator, server *Server) {
+	server.currGatherTime = time.Now().Unix()
+
+	// first time gather data from the past 10s
+	if server.lastGatherTime == 0 {
+		server.lastGatherTime = time.Now().Unix() - 10
+	}
+
+	chartRequestURL, err := url.Parse(server.Url + chartsURL)
+	if err != nil {
+		acc.AddError(Errorf(err, "error parsing charts url : %s", server.Url+chartsURL))
+		return
+	}
+
+	chartRespBody, err := n.sendRequest(chartRequestURL, server)
+	if err != nil {
+		acc.AddError(Errorf(err, "error getting netdata charts, url : %s", chartRequestURL.String()))
+		return
+	}
+
+	var chartResp ResponseCharts
+	if err := json.Unmarshal([]byte(chartRespBody), &chartResp); err != nil {
+		acc.AddError(Errorf(err, "error unmarshalling netdata charts"))
+		return
+	}
+
+	for _, chart := range chartResp.Charts {
+		dataStringURL := n.buildChartDataURL(server, chart)
+
+		dataRequestURL, err := url.Parse(dataStringURL)
+
+		if err != nil {
+			acc.AddError(Errorf(err, "error parsing chart data url : %s", dataStringURL))
+			continue
+		}
+
+		dataRespBody, err := n.sendRequest(dataRequestURL, server)
+		if err != nil {
+			acc.AddError(Errorf(err, "error getting netdata chart data, url : %s", dataRequestURL.String()))
+			continue
+		}
+
+		var dataResp ResponseData
+		if err := json.Unmarshal([]byte(dataRespBody), &dataResp); err != nil {
+			acc.AddError(Errorf(err, "error unmarshalling netdata chart data"))
+			continue
+		}
+
+		dimensions := strings.Join(dataResp.DimensionNames, ",")
+
+		var measurement string
+		if n.PrefixChartIDByHostname {
+			measurement = "netdata." + chartResp.Hostname + "." + chart.ID
+		} else {
+			measurement = "netdata." + chart.ID
+		}
```

Comment:
What is the use-case for inserting the hostname there? It is also given in the tag `hostname`...

Reply:
Mandatory.